### PR TITLE
[Log Collector] Add sidecar support

### PR DESCRIPTION
**Pull request by Tankilevitch**

### CODE REVIEW

File: mlrun/config.py [hunk 1]
```diff
@@ -467,6 +467,30 @@
...
+    "log_collector": {
+        "address": "localhost:8282",
+        # log collection mode can be one of: "sidecar", "legacy", "best-effort"
+        # "sidecar" - use the sidecar to collect logs
+        # "legacy" - use the legacy log collection method (logs are collected straight from the pod)
+        # "best-effort" - use the sidecar, but if for some reason it's not available use the legacy method
+        # note that this mode also effects the log querying method as well, meaning if the mode is "best-effort"
+        # the log query will try to use the sidecar first and if it's not available it will use the legacy method
+        # TODO: once this is changed to "sidecar" by default, also change in common_fixtures.py
+        "mode": "legacy",
```
Comment (Tankilevitch) [inline mlrun/config.py:479]: Need to change to sidecar

Comment (liranbg) [reply mlrun/config.py:479]: the opposite i think. retain as legacy for when running under docker

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -19,6 +19,7 @@ on:
     branches:
     - development
     - '[0-9]+.[0-9]+.x'
+    - log-collector-side-car
```
Comment (liranbg) [inline .github/workflows/ci.yaml:22]: can be remove

File: mlrun/db/httpdb.py [hunk 3]
```diff
@@ -205,6 +210,10 @@ def api_call(
             )
 
         try:
+            if stream:
+                return self.session.stream_request(
+                    method, url, to_stdout=to_stdout, timeout=timeout, **kw
+                )
```
Comment (TomerShor) [inline mlrun/db/httpdb.py:216]: Does anybody use it (client/sdk/ui)?
I couldn't find a call with `stream=True`.
If not we can remove it

File: mlrun/utils/http.py [hunk 4]
```diff
@@ -134,6 +134,25 @@ def request(self, method, url, **kwargs):
...
+        response_body = ""
+        with super().request(method, url, stream=True, **kwargs) as resp:
+            for line in resp.iter_lines():
+                if line:
+                    if to_stdout:
+                        print(line.decode())
+                    else:
+                        response_body += line.decode()
+        return response_body
+
```
Comment (TomerShor) [inline mlrun/utils/http.py:155]: Same, if not called from `api_call` can be removed.

**Review by TomerShor — CHANGES_REQUESTED** [review]

**Pull request merged**